### ISSUE TITLE
Add diagnostic for when a TypeVar references itself as a default

### ISSUE THREAD

**Issue by MeGaGiGaGon** [astral-sh/ty#3235]
This is a diagnostic from BasedPyright that helps explain why Unknowns can appear in defaults:
```py
from typing import reveal_type

class A[T = T]: ...  # Type of "T" could not be determined because it refers to itself
reveal_type(A())  # Type of "A()" is "A[Unknown]"
```
BasedPyright doesn't have an error for this trickier version, but it could also be included:
```py

from typing import reveal_type

class A[T = A]: ...  # Implicitly self-referential through the default value
```
A good name could be `cyclic-type-alias-default`
Also note that unlike `cyclic-type-alias-definition`, here putting the TypeVar inside the generic of another class does not fix the issue, as the `T` should be fully removed from the built type's signature:
```py
class A[T = list[T]]: ...
# Would default to A[list[list[list[...]]]]
```

cc astral-sh/ruff#24455
